rpm/build/SPECS/jenkins.spec

```diff
@@ -76,30 +76,57 @@ rm -rf "%{buildroot}"
 /usr/sbin/useradd -g %{name} -s /bin/false -r -c "@@SUMMARY@@" \
 	-d "%{workdir}" %{name} &>/dev/null || :
 
+  # Used to decide later if we should perform a chown in case JENKINS_INSTALL_SKIP_CHOWN is false
+  # Check if a previous installation exists, if so check the JENKINS_HOME value and existing owners of work, log and cache dir, need to to this check
+  # here because the %files directive overwrites folder owners, I have not found a simple way to make the
+  # files directive to use JENKINS_USER as owner.
+  if [ -f "/etc/sysconfig/%{name}" ]; then
+      . /etc/sysconfig/%{name}
+      touch /tmp/cacheowner
+      ls -ld /var/cache/%{name} | awk 'NR==1 {print $3}' > /tmp/cacheowner
+      touch /tmp/logowner
+      ls -ld /var/log/%{name} | awk 'NR==1 {print $3}' >  /tmp/logowner
+      touch /tmp/workdirowner
+      ls -ld ${JENKINS_HOME:-%{workdir}}| awk 'NR==1 {print $3}'  > /tmp/workdirowner
+  fi
+
 %post
 /sbin/chkconfig --add %{name}
 
-# If we have an old hudson install, rename it to jenkins
-if test -d /var/lib/hudson; then
-    # leave a marker to indicate this came from Hudson.
-    # could be useful down the road
-    # This also ensures that the .??* wildcard matches something
-    touch /var/lib/hudson/.moving-hudson
-    mv -f /var/lib/hudson/* /var/lib/hudson/.??* /var/lib/%{name}
-    rmdir /var/lib/hudson
-    find /var/lib/%{name} -user hudson -exec chown %{name} {} + || true
-fi
-if test -d /var/run/hudson; then
-    mv -f /var/run/hudson/* /var/run/%{name}
-    rmdir /var/run/hudson
-fi
-
-# Ensure the right ownership on files
+# Ensure the right ownership on files only if not owned by JENKINS_USER and JENKINS_USER
+# != %{name}, namely all cases but the default one (configured for %{name} owned by %{name})
+# In any case if JENKINS_INSTALL_SKIP_CHOWN is true we do not chown anything to maintain
+# the existing semantics
 . /etc/sysconfig/%{name}
 if test x"$JENKINS_INSTALL_SKIP_CHOWN" != "xtrue"; then
-   chown -R ${JENKINS_USER:-%{name}} /var/cache/%{name}
-   chown -R ${JENKINS_USER:-%{name}} /var/log/%{name}
-   chown -R ${JENKINS_USER:-%{name}} ${JENKINS_HOME:-%{workdir}}
+    if [ -f "/tmp/cacheowner" ]; then
+      cacheOwner=$(cat /tmp/cacheowner)
+      rm -f /tmp/cacheowner
+    fi
+    if  [ "${cacheOwner:-%{name}}" != "${JENKINS_USER:-%{name}}" ] ; then
+        chown -R ${JENKINS_USER:-%{name}} /var/cache/%{name}
```

Comment:
Is there a reasonable outlet for log output during this phase? If so, probably not a bad idea to do so. If not, also fine.